### PR TITLE
Support providing the CLI version

### PR DESCRIPTION
**Pull request by yahavi**
- [x] This pull request is created in the [jfrog/jenkins-jfrog-plugin](https://github.com/jfrog/jenkins-jfrog-plugin) repository.

Allow providing the JFrog CLI version in the ArtifactoryInstaller and in the ReleasesInstaller.
* Add help tags
* Add form validators

image

image

### CODE REVIEW

File: src/main/java/io/jenkins/plugins/jfrog/ReleasesInstaller.java [hunk 1]
```diff
@@ -57,5 +49,11 @@ public String getDisplayName() {
         public boolean isApplicable(Class<? extends ToolInstallation> toolType) {
             return toolType == JfrogInstallation.class;
         }
+
+        @POST
+        @SuppressWarnings("unused")
+        public FormValidation doCheckVersion(@QueryParameter String version) {
```
Comment (eyalbe4) [inline src/main/java/io/jenkins/plugins/jfrog/ReleasesInstaller.java:55]: doCheckVersion --> doValidateVersion

File: src/main/webapp/help/ArtifactoryInstaller/help-serverId.html [hunk 2]
```diff
@@ -0,0 +1,3 @@
+<div>
+    Server ID of your JFrog instance, which configured in Manage Jenkins | Configure System.
```
Comment (eyalbe4) [inline src/main/webapp/help/ArtifactoryInstaller/help-serverId.html:2]: ...which configured in Manage Jenkins | Configure System....
-->
...which is configured in 'Manage Jenkins | Configure System'....

**Review by eyalbe4 — CHANGES_REQUESTED** [review]

File: src/main/java/io/jenkins/plugins/jfrog/ArtifactoryInstaller.java [hunk 3]
```diff
@@ -84,5 +127,20 @@ public boolean isApplicable(Class<? extends ToolInstallation> toolType) {
         public List<JFrogPlatformInstance> getServerIds() {
             return JFrogPlatformBuilder.getJFrogPlatformInstances();
         }
+
+        @POST
+        @SuppressWarnings("unused")
+        public FormValidation doCheckRepository(@QueryParameter String repository) {
```
[the remark this answers is not in the document]
Comment (yahavi) [reply src/main/java/io/jenkins/plugins/jfrog/ArtifactoryInstaller.java:133]: The "doCheck[varialbe]" is a convention in Jenkins forms and can't be changed

File: src/main/java/io/jenkins/plugins/jfrog/ArtifactoryInstaller.java [hunk 4]
```diff
@@ -84,5 +127,20 @@ public boolean isApplicable(Class<? extends ToolInstallation> toolType) {
...
+            if (StringUtils.isBlank(repository)) {
+                return FormValidation.error("Required");
+            }
+            return FormValidation.ok();
+        }
+
+        @POST
+        @SuppressWarnings("unused")
+        public FormValidation doCheckVersion(@QueryParameter String version) {
```
[the remark this answers is not in the document]
Comment (yahavi) [reply src/main/java/io/jenkins/plugins/jfrog/ArtifactoryInstaller.java:142]: The "doCheck[varialbe]" is a convention in Jenkins forms and can't be changed

[the remark this answers is not in the document]
Comment (eyalbe4) [reply src/main/java/io/jenkins/plugins/jfrog/ArtifactoryInstaller.java:141]: NP.

File: src/main/java/io/jenkins/plugins/jfrog/ReleasesInstaller.java (hunk 1 above)
Comment (yahavi) [reply src/main/java/io/jenkins/plugins/jfrog/ReleasesInstaller.java:55]: The "doCheck[varialbe]" is a convention in Jenkins forms and can't be changed

**Pull request merged**